presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -1059,11 +1060,12 @@ public GroupedExecutionProperties visitTableScan(TableScanNode node, Void contex
             }
             List<ConnectorPartitionHandle> partitionHandles = nodePartitioningManager.listPartitionHandles(session, tablePartitioning.get().getPartitioningHandle());
             boolean recoveryEligible = metadata.getConnectorCapabilities(session, node.getTable().getConnectorId()).contains(SUPPORTS_REWINDABLE_SPLIT_SOURCE);
+            boolean useful = isGroupedExecutionForAllEligibleTableScansEnabled(session);
             if (ImmutableList.of(NOT_PARTITIONED).equals(partitionHandles)) {
-                return new GroupedExecutionProperties(false, false, ImmutableList.of(), 1, recoveryEligible);
+                return new GroupedExecutionProperties(false, useful, ImmutableList.of(), 1, recoveryEligible);
```

Comment:
I feel `GroupedExecutionProperties#useful` should be renamed. But don't have to do it in this PR.